app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt

```diff
@@ -0,0 +1,156 @@
+package com.arielwang.workoutlogger.features.exercise.ui.screen
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.arielwang.workoutlogger.R
+import com.arielwang.workoutlogger.features.component.ScaffoldWithTopBar
+import com.google.accompanist.insets.LocalWindowInsets
+import com.google.accompanist.insets.toPaddingValues
+
+@Composable
+fun ExerciseScreen(
+    uiState: ExerciseView.State = ExerciseView.State(),
+    onAction: (ExerciseView.Action) -> Unit
+) {
+    ScaffoldWithTopBar(
+        title = R.string.ExerciseScreen_topbarTitle,
+        iconContentDescription = R.string.ExerciseScreen_topbarIconButtonContentDescription,
+        onAction = { onAction(ExerciseView.Action.GoBackToPreviousPage)},
+        modifier = Modifier.systemBarsPadding()
+    ) {
+        ExerciseContentConstrainLayout(uiState = uiState, onAction = onAction)
+    }
+
+}
+
+@Composable
+fun ExerciseContentConstrainLayout(
+    uiState: ExerciseView.State,
+    onAction: (ExerciseView.Action) -> Unit
+) {
+    ConstraintLayout(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 16.dp)
+    ) {
+        val (button, cardListSection) = createRefs()
+
+        ExerciseCardList(
+            uiState = uiState,
+            onAction = onAction,
+            cardListConstrainLayout = Modifier.constrainAs(cardListSection) {
+                top.linkTo(parent.top)
+                bottom.linkTo(button.top)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            }
+        )
+
+        ExerciseScreenButton(
+            onAction = onAction,
+            buttonConstraintLayout = Modifier.constrainAs(button) {
+                bottom.linkTo(parent.bottom)
+            }
+        )
+    }
+}
+
+@Composable
+fun ExerciseCardList(
+    uiState: ExerciseView.State,
+    onAction: (ExerciseView.Action) -> Unit,
+    cardListConstrainLayout: Modifier
+) {
+    LazyColumn(
+        modifier = cardListConstrainLayout
+    ) {
+        item {Spacer(modifier = Modifier.height(30.dp))}
```

Comment:
We always want to use multiple of 8 in these dimensions like 16.dp, 24.dp, 32.dp
https://uxplanet.org/everything-you-should-know-about-8-point-grid-system-in-ux-design-b69cb945b18d